src/core/core/rpc_engine.cpp

```diff
@@ -846,14 +850,28 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
+                                /*
+                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
```

Comment:
You are right, I was thinking about the async case - there are indeed issues about the sync case. Let's figure out a solution to it - the retry is important otherwise you will always have an error in your first try when there is a reconfiguration or it is the code start from the client perspective.